src/ducks/Signals/SignalMaster.js

```diff
@@ -37,19 +37,38 @@ export class SignalMaster extends React.PureComponent {
     }
     const { step, trigger } = this.state
     const currentTrigger = trigger || (this.props.trigger || {}).trigger
-    const formProps = currentTrigger
+    let formProps = currentTrigger
       ? mapTriggerToFormProps(currentTrigger)
       : undefined
     const meta = {
       title: currentTrigger ? currentTrigger.title : 'Any',
       description: currentTrigger ? currentTrigger.description : 'Any'
     }
+
+    let metaFormSettings = { ...this.props.metaFormSettings }
+
+    if (!this.props.metaFormSettings && this.props.asset) {
+      metaFormSettings = {
+        ...metaFormSettings,
+        ...{
```

Comment:
@GarageInc Why there is a need for spreading this object? Thanks in advance

Reply:
My mistake, metaFormSettings  is a wrong. Need to change into formSettings mutable object.